File: Pkdd/pkdd-ng-client/src/app/help/models/answer.ts
import { TimeTrack } from './../../models/common/time-track';
import { IEntity } from './../../models/entities/interfaces/iEntity';
import { IssueUserInfo } from './issue-user-info';
import { PkddUser } from 'src/app/models/auth/pkdd-user';

export class Answer implements IEntity {
    id: number;
    isDeleted = false;
    timeTrack: TimeTrack = new TimeTrack(new Date(), new Date(), new Date());
    issueId: number;
    answerText = '';
    user: IssueUserInfo;
    order: number;
    constructor(user: PkddUser, order: number, issueId: number) {
        this.user = new IssueUserInfo(user);
        this.order = order;
        this.issueId = issueId;
    }
}

File: Pkdd.Models/Help/Answer.cs
using Pkdd.Abstractions.Entity;
using Pkdd.Models.Users;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Pkdd.Models.Help
{
    public class Answer: EntityBase<Answer>
    {
        public int IssueId { get; set; }

        public string AnswerText { get; set; }

        public IssueUserInfo User { get; set; }

        public int Order { get; set; }

        public override Answer Update(Answer entity)
        {
            AnswerText = entity.AnswerText;
            Order = entity.Order;
            IsDeleted = entity.IsDeleted;
            return this;
        }
    }
}
